crates/bevy_ecs/src/query/iter.rs

```diff
@@ -173,24 +174,28 @@ impl<'w, 's, D: QueryData, F: QueryFilter> QueryIter<'w, 's, D, F> {
         let entities = archetype.entities();
         for index in indices {
             // SAFETY: Caller assures `index` in range of the current archetype.
-            let archetype_entity = entities.get_unchecked(index);
+            let archetype_entity = unsafe { entities.get_unchecked(index) };
             // SAFETY: set_archetype was called prior.
             // Caller assures `index` in range of the current archetype.
-            if !F::filter_fetch(
-                &mut self.cursor.filter,
-                archetype_entity.id(),
-                archetype_entity.table_row(),
-            ) {
+            if unsafe {
```

Comment:
Same with the match, having an unsafe block as the conditional in an if expression is not great for readability.